packages/gatsby-source-wordpress/src/utils/fetch-graphql.ts

```diff
@@ -302,37 +327,41 @@ const handleFetchErrors = async ({
       },
     })
   }
-  if (e.message.includes(`Request failed with status code 50`)) {
-    const {
-      requestConcurrency,
-      previewRequestConcurrency,
-    } = store.getState().gatsbyApi.pluginOptions.schema
+  if (
+    e.message.includes(`Request failed with status code 50`) &&
+    [`502`, `503`, `504`].includes(e.message)
+  ) {
+    if (`message` in e) {
+      console.error(formatLogMessage(new Error(e.message).stack))
+    }
 
-    console.error(e)
     reporter.panic({
       id: CODES.WordPress500ishError,
       context: {
         sourceMessage: formatLogMessage(
-          [
-            `Your wordpress server at ${bold(url)} appears to be overloaded.`,
-            `\nTry reducing the ${bold(
-              `requestConcurrency`
-            )} for content updates or the ${bold(
-              `previewRequestConcurrency`
-            )} for previews:`,
-            `\n{
-  resolve: 'gatsby-source-wordpress',
-  options: {
-    schema: {
-      requestConcurrency: 5, // currently set to ${requestConcurrency}
-      previewRequestConcurrency: 2, // currently set to ${previewRequestConcurrency}
-    }
-  },
-}`,
-            `\nThe ${bold(
-              `GATSBY_CONCURRENT_REQUEST`
-            )} environment variable is no longer used to control concurrency.\nIf you were previously using that, you'll need to use the settings above instead.`,
-          ].join(`\n`),
+          `Your WordPress server at ${bold(url)} appears to be overloaded.
+
+${getLowerRequestConcurrencyOptionMessage()}`,
+          { useVerboseStyle: true }
+        ),
+      },
+    })
+  } else if (e.message.includes(`Request failed with status code 500`)) {
+    reporter.panic({
+      id: CODES.WordPress500ishError,
+      context: {
+        sourceMessage: formatLogMessage(
+          `${e.message}
+
+Your WordPress server is either overloaded or encountered a PHP error.
+
+${errorContext}
+
+Enable WP_DEBUG, WP_DEBUG_LOG, and WPGRAPHQL_DEBUG, run another build and then check your WordPress instance's debug.log file.
+
+If you don't see any errors in debug.log:
+
+${getLowerRequestConcurrencyOptionMessage()}`,
```

Comment:
This is largely what inspired [my little experiment this morning](https://gatsbyjs.slack.com/archives/C01AQRY95FV/p1614874340081300) because I've had to deal with this as well and the dilemma just doesn't sit right with me. Maybe I'm overly fussy about it but I think something like this could really make our source plugin code look cleaner.

So, it's not a suggestion for this PR, but I'm wondering if you think a utility like that would be useful for us.

Reply:
@sslotsky I really like it! I think the name could use some workshopping - prettyPrint makes me think of prettier where it'll be doing some formatting rather than handling indentation. Maybe `prettyIndent`? Anyway I think it's a great idea for a helper and I agree it would make things a lot cleaner. I've seen folks handle this in a few ways and none of them sit right to me. Your helper is the cleanest I've seen :+1: